stream.go

```diff
@@ -66,45 +72,182 @@ func (f *File) NewStreamWriter(sheet string) (*StreamWriter, error) {
 		File:    f,
 		Sheet:   sheet,
 		SheetID: sheetID,
+		replaceMap: map[string][]byte{
+			"XMLName": []byte{},
+		},
 	}
+	rsw.encoder = xml.NewEncoder(&rsw.SheetData)
 	rsw.SheetData.WriteString("<sheetData>")
 	return rsw, nil
 }
 
-// SetRow writes an array to streaming row by given worksheet name, starting
-// coordinate and a pointer to array type 'slice'. Note that, cell settings
-// with styles are not supported currently and after set rows, you must call the
-// 'Flush' method to end the streaming writing process. The following
-// shows the supported data types:
+// SetTable creates an Excel table for the StreamWriter. This function must be
+// called before the first SetRow.
 //
-//    int
-//    string
-//
-func (sw *StreamWriter) SetRow(axis string, slice interface{}) error {
+// The first SetRow will create the column names. These Column names must be
+// strings or an error will be returned.
+func (sw *StreamWriter) SetTable(format string) error {
+	formatSet, err := parseFormatTableSet(format)
+	if err != nil {
+		return err
+	}
+
+	tableID := sw.File.countTables() + 1
+
+	name := formatSet.TableName
+	if name == "" {
+		name = "Table" + strconv.Itoa(tableID)
+	}
+
+	sw.table = &xlsxTable{
+		XMLNS:       NameSpaceSpreadSheet,
+		ID:          tableID,
+		Name:        name,
+		DisplayName: name,
+		TableStyleInfo: &xlsxTableStyleInfo{
+			Name:              formatSet.TableStyle,
+			ShowFirstColumn:   formatSet.ShowFirstColumn,
+			ShowLastColumn:    formatSet.ShowLastColumn,
+			ShowRowStripes:    formatSet.ShowRowStripes,
+			ShowColumnStripes: formatSet.ShowColumnStripes,
+		},
+	}
+	return nil
+}
+
+// SetRow writes an array to stream rows by giving a worksheet name, starting
+// coordinate and a pointer to an array of values. If styles is non-nil, then
+// the styles must be the same size as the values and will be applied to each
+// corresponding cell. Note that you must call the 'Flush' method to end the
+// streaming writing process.
+func (sw *StreamWriter) SetRow(axis string, values []interface{}, styles []int) error {
```

Comment:
Create a new exported type `Cell` that can be wrapped by the interface, type `Cell` include integer type StyleID filed, instead of the third parameter `styles`.

Reply:
This is now done with 443da043d89438707cde812c7dcb3efcc7bc0058.